semperfiwebdesign/all-in-one-seo-pack
# Add PHPCompatibility sniffs to PHPCS & fix issues

Issue #2851 (& #2834)

## Proposed changes

This PR adds the PHPCompatibility sniffs to our PHPCS configuration and fixes any remaining compatibility issues in our code.

Also, I made a few other changes:

- Added PHPCS, WPCS & PHPCS Composer Installer as Composer dependencies
- Added Composer and VSCode folders to .gitignore
- Renamed "inc/compatability" to "inc/compatibility" to fix the typo
- Added a separate Grunt task set for Travis that doesn't run phpcbf, uglify & cssmin (these are useless when testing builds)

## Types of changes

- Bugfixes (non-breaking changes which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Improves existing functionality
- Improves existing code

## Checklist


## Testing instructions

There isn't much that can be tested here. I'd recommend having a look at the code changes, then installing the PR and just having a brief look to see if all looks good. 

Navigate through the menus, go to the meta box and make sure WP_DEBUG is enabled. You'll also want to create a Post, enter a description with special characters and make sure those still get encoded the same as before.

## Further comments

In addition to ensuring our PHP cross-version compatibility, this PR will also make it much easier to setup the development environment locally as PHPCS and WPCS no longer need to be downloaded and configured manually.
